File: js/object_stuff.js
"use strict";

class PondAnimal {

  constructor() {
    this.sound = "";
    this.flies = false;
  }

  describe() {
    console.log(`I make ${this.sound} sounds`);
    if (this.flies) {
      console.log("I can fly!");
    } else {
      console.log("I can't fly :(");
    }
  }
}

class Frog extends PondAnimal {
  constructor() {
    super();
    this.sound = "ribbit";
  }
}

class Duck extends PondAnimal {
  constructor() {
    super();
    this.sound = "quack";
    this.flies = true;
  }
}

let frog = new Frog();
frog.describe();

let duck = new Duck();
duck.describe();

File: ruby/object_stuff.rb
class PondAnimal
  attr_accessor :sound, :flies

  def initialize
    self.sound = ""
    self.flies = false
  end

  def describe
    puts("I make #{sound} sounds")
    if flies
      puts("I can fly!")
    else
      puts("I can't fly :(")
    end
  end
end

class Frog < PondAnimal
  def initialize
    super
    self.sound = "ribbit"
  end
end

class Duck < PondAnimal
  def initialize
    super
    self.sound = "quack"
    self.flies = true
  end
end

frog = Frog.new
frog.describe

duck = Duck.new
duck.describe
